src/openms/source/FILTERING/DATAREDUCTION/FeatureOverlapFilter.cpp

```diff
@@ -55,32 +160,50 @@ namespace OpenMS
     float minRT = fmap.getMinRT();
     float maxRT = fmap.getMaxRT();
 
+    // build quadtree with all features
     quadtree::Box<float> fullExp(minMZ-1,minRT-1,maxMZ-minMZ+2,maxRT-minRT+2);
     auto quadtree = quadtree::Quadtree<Feature*, decltype(getBox)>(fullExp, getBox);
     for (auto& f : fmap)
         quadtree.add(&f);
 
+    // if we check for overlapping traces we need a faster lookup structure
+    FeatureBoundsMap fbm;
+    if (check_overlap_at_trace_level)
+    {
+      fbm = getFeatureBounds(fmap);
+    }
+
     std::unordered_set<Size> removed_uids;
     for (const auto& f : fmap)
     {
       if (removed_uids.count(f.getUniqueId()) == 0)
       {
         for (const auto& overlap : quadtree.query(getBox(&f)))
         {
-          if (overlap != &f)
+          if ((overlap != &f))
           {
-            removed_uids.insert(overlap->getUniqueId());
-            overlap->setOverallQuality(-1.); // used to filter
+            // Because feature boundaries might be large and lead to many overlapps, we (optionally) also can check if the boundaries of traces overlap
+            bool is_true_overlap = true;
+            if (check_overlap_at_trace_level)
+            {            
+              is_true_overlap = tracesOverlap(f, *overlap, fbm) ? true : false;
+            }
+
+            if (is_true_overlap)
+            {
+              removed_uids.insert(overlap->getUniqueId());
+              overlap->setOverallQuality(-1.); // used to filter
```

Comment:
magic constant alert!
Please use a real constant, e.g.
```
  constexpr int NO_OVERLAP = -1;
```
```suggestion
              overlap->setOverallQuality(NO_OVERLAP); // used to filter
```